### ISSUE TITLE
react-plugin: "immutability" rule flags Signal.value mutation in non-React use*-prefixed hooks (e.g. Qwik) as invalid

### ISSUE THREAD

**Issue by ixcans** [oxc-project/oxc#26094]
### What happened

Running oxlint with `--react-plugin` against a Qwik (https://qwik.dev) codebase produces `react(immutability)` warnings on code that is completely correct and idiomatic for that framework:

```
warning react(immutability): This value cannot be modified
help: Modifying a value returned from a hook is not allowed. Consider moving the modification into the hook where the value is constructed
```

### Minimal reproduction

```ts
import { useSignal, $ } from "@qwik.dev/core";

export const useVote = () => {
  const isVoting = useSignal(false);

  const handleVote$ = $(async () => {
    if (isVoting.value) return;
    isVoting.value = true; // <-- flagged: "Modifying a value returned from a hook is not allowed"
    // ...
  });

  return { handleVote$ };
};
```

`oxlint --react-plugin repro.ts` (v1.80.0) reports:

```
repro.ts:7:5: warning react(immutability): This value cannot be modified help: Modifying a value returned from a hook is not allowed. Consider moving the modification into the hook where the value is constructed
```

### Why this is a false positive

The rule appears to pattern-match any function whose name is prefixed `use` and treat its return value as immutable, the way a React hook's returned state actually should be (React expects a setter call, never a direct mutation).

Qwik deliberately mirrors React's `use*` naming convention for developer familiarity, but `useSignal()` returns a `Signal<T>` object whose `.value` property is *designed* to be mutated directly — that assignment is how Qwik's reactivity system observes the change (see https://qwik.dev/docs/components/state/#usesignal). There is no setter function; direct assignment to `.value` is the correct, idiomatic, and only way to update it, both inside and outside the hook that created it.

Because the `use*` naming convention is shared with other frameworks that don't follow React's own hooks contract, this rule seems to assume any codebase it runs against is React specifically, with no way to opt a project out short of disabling `immutability` entirely — which also loses whatever legitimate value the rule has for an actual React project.

### Environment

- oxlint 1.80.0 (`npx oxlint@latest --react-plugin`)
- Reproduced directly against a real app (a Qwik 2 codebase) — 75 warnings from `--react-plugin` across the whole `src/` tree, essentially all in this same family (`immutability` misfiring on `Signal.value` writes inside `use*`-prefixed hooks).

### What I'd expect

Ideally the rule would only apply within an actual React component/hook body (e.g. gated on the configured JSX runtime / a real `react` import, similar to how other react-hooks lints avoid firing on non-React code), or at minimum there would be a documented way to scope/disable just this heuristic without losing the rest of the react-plugin's rules.

I found a related-but-different existing tracking issue, #1022, which covers ESLint-parity for `react/jsx-uses-react`-style checks, but it doesn't mention this "any `use*`-prefixed function's return value is assumed immutable" heuristic specifically.

There's also a `react(purity)` warning on a bare `Date.now()` call inside a `$()`-wrapped Qwik callback that may be a related mis-detection of what counts as "render" code — flagging it here for awareness, though I'm less confident that one isn't also a legitimate general-purpose warning outside the Qwik-specific case above.

**Comment by leaysgur** [comment]
`--react-plugin` enables React-specific linting semantics.
And Qwik is a different framework, so the React plugin is not intended or recommended for Qwik projects.

If you still want to use some React plugin rules, you can disable individual rules without disabling the rest of the plugin.

Please see:

- Configuring individual rules
  - https://oxc.rs/docs/guide/usage/linter/config.html#configure-rules
- File-pattern overrides
  - https://oxc.rs/docs/guide/usage/linter/config.html#apply-configuration-by-file-pattern
- CLI --allow
  - https://oxc.rs/docs/guide/usage/linter/cli.html#allowing-denying-multiple-lints
